docker/service.go

```diff
@@ -273,3 +279,13 @@ func (s *Service) Containers() ([]project.Container, error) {
 
 	return result, nil
 }
+
+func (s *Service) SpecificiesHostPort() bool {
+	for _, port := range s.serviceConfig.Ports {
+		if strings.ContainsRune(port, ':') {
+			return true
+		}
+	}
```

Comment:
Refer to https://github.com/docker/libcompose/blob/master/docker/convert.go#L60 for the proper way to parse a port spec.